recipes/boost/all/conanfile.py

```diff
@@ -703,6 +730,22 @@ def _run_bcp(self):
                 self.run(command)
 
     def build(self):
+        if tools.cross_building(self.settings, skip_x64_x86=True):
+            # When cross building, do not attempt to run the test-executable (assume they work)
+            tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "libs", "stacktrace", "build", "Jamfile.v2"),
+                                  "$(>) > $(<)",
+                                  "echo \"\" > $(<)", strict=False)
+        # Older clang releases require a thread_local variable to be initialized by a constant value
+        tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "boost", "stacktrace", "detail", "libbacktrace_impls.hpp"),
+                              "/* thread_local */", "thread_local", strict=False)
+        tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "boost", "stacktrace", "detail", "libbacktrace_impls.hpp"),
+                              "/* static __thread */", "static __thread", strict=False)
+        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) < 6:
+            tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "boost", "stacktrace", "detail", "libbacktrace_impls.hpp"),
+                                  "thread_local", "/* thread_local */")
+            tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "boost", "stacktrace", "detail", "libbacktrace_impls.hpp"),
+                                  "static __thread", "/* static __thread */")
```

Comment:
seems odd to be undoing the previous release

Maybe?

```suggestion
        # Older clang releases require a thread_local variable to be initialized by a constant value
        if not (self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) < 6):
            tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "boost", "stacktrace", "detail", "libbacktrace_impls.hpp"),
                                  "/* thread_local */", "thread_local", strict=False)
            tools.replace_in_file(os.path.join(self.source_folder, self._source_subfolder, "boost", "stacktrace", "detail", "libbacktrace_impls.hpp"),
                                  "/* static __thread */", "static __thread", strict=False)
```

Reply:
This is done for a purpose.
The boost conan recipe is a *bad recipe* in that it modifies and creates sources in the sources folder.
They are not copied to the build folder, or there is not an out-of-tree build going on.
That's why I try to revert the state of the build script to an initial state before patching.
This way when building for e.g. clang-5 and gcc-9, the patch is applied for clang but not for gcc-9.

Reply:
Then probably it is easier if we remove the `no_copy_source` from the recipe class. Simpler will be better in the long run, although it will take more time to build (but, hey! This is Conan, build once and use a thousand times 😄 ). 

Not for this PR, but something to consider.